### PR TITLE
Implement a new documentation bot

### PR DESCRIPTION
**Pull request by maxsxu**
Master issue: https://github.com/apache/pulsar/issues/15797

## Motivation

As discussed at https://lists.apache.org/thread/tz7x73lnkvcg3vqmb9r26s4p6md5sb8y

## Modification

Created a new folder named `docbot`, which contains the implementation.

**Supported Features**

- Automatic labeling according to PR body when created or updated
- Able to update PR body automatically when labels is manually selected
- Configurable parameters including which labels want to watch, whether allow empty labels, etc

**Directory Structure**
This `docbot` is implemented by [Golang](https://go.dev/) with following directory structure:
```
├── pkg/logger/logger.go     // Contains a simple logger for logging in the actor details
├── README.md
├── action.yml              // The action configuration file
├── go.mod
└── main.go                // The main logic is here
```

> Note: The old folder `doc-label-check` will be removed after this PR merged.

## Validation
@Anonymitaet has helped validate the required features on the [prototype repository](https://github.com/maxsxu/action-labeler).

### CODE REVIEW

File: docbot/action.yml
Comment (Anonymitaet) [inline docbot/action.yml:3]: how about adding your GitHub ID so that others can find you easily if they have any questions?

**Review by Anonymitaet — APPROVED** [review]
@maxsxu thanks for your great contribution! We've tested and it works well. Approve from the non-technical side.

**Review by urfreespace — APPROVED** [review]

File: docbot/action.yml
Comment (maxsxu) [reply docbot/action.yml]: Make sense, the author's email is added.

**Comment by lhotari** [comment]
> @maxsxu thanks for your great contribution! We've tested and it works well. Approve from the non-technical side.

@Anonymitaet how was this tested? all jobs fail in apache/pulsar currently after merging https://github.com/apache/pulsar/pull/16027

File: docbot/README.md
Comment (michaeljmarshall) [inline docbot/README.md:29]: Did you meant to leave this reference to your personal github repo?

File: docbot/README.md
Comment (michaeljmarshall) [inline docbot/README.md:38]: Same comment.

**Comment by Anonymitaet** [comment]
> > @maxsxu thanks for your great contribution! We've tested and it works well. Approve from the non-technical side.
> 
> @Anonymitaet how was this tested? all jobs fail in apache/pulsar currently after merging [apache/pulsar#16027](https://github.com/apache/pulsar/pull/16027)

@lhotari we've tested in the prototype repo https://github.com/maxsxu/action-labeler manually, like opening a PR and giving it different labels to watch how the bot run under different situations.

@maxsxu has fixed the issues yesterday (you've merged the PR)

Are there any other (automatic) tests we should do?

File: docbot/README.md
Comment (Anonymitaet) [reply docbot/README.md:29]: I guess so. Any concern on this?

Comment (michaeljmarshall) [reply docbot/README.md:29]: My only concern is that the README is incorrect. We should fix it so that it documents how to use the action in this repository.

**Comment by michaeljmarshall** [comment]
> Able to update PR body automatically when labels is manually selected

I think we should remove this functionality. If the label is manually added, there is no need for the PR body to have the check box. The end goal is to have the right label on the PR, and if the label is already added, the bot doesn't need to do anything.

Further, on this PR https://github.com/apache/pulsar/pull/16045, it appears that the bot updated my PR's body 3 times to add the same string (look in the PR body history).

**Comment by Anonymitaet** [comment]
> I think we should remove this functionality. If the label is manually added, there is no need for the PR body to have the check box. The end goal is to have the right label on the PR, and if the label is already added, the bot doesn't need to do anything.

@michaeljmarshall this functionality is useful and needed.

1. Check box
  Contributors without committer privilege can label a PR only through a check box in a PR description. The check box is for their sake. 

2. Bot
  Some contributors do not provide doc-related labels even if they get notifications from doc Bot and comments from writers.
Failed doc label check is another way to remind them please show care to docs. A project is nothing if there are no docs.

> Further, on this PR https://github.com/apache/pulsar/pull/16045, it appears that the bot updated my PR's body 3 times to add the same string (look in the PR body history).

@maxsxu has fixed this bug in https://github.com/apache/pulsar/pull/16060

File: docbot/README.md
Comment (Anonymitaet) [reply docbot/README.md:29]: Make sense! @maxsxu will update the readme.

Comment (maxsxu) [reply docbot/README.md:29]: Fixed in https://github.com/apache/pulsar-test-infra/pull/48

**Pull request merged**
